bind/uniffi/src/errors.rs

```diff
@@ -0,0 +1,107 @@
+use std::sync::Arc;
+use std::{any::type_name, fmt::Debug};
+use thiserror::Error;
+use web5::apid::credentials::presentation_definition::PexError;
+use web5::apid::credentials::CredentialError;
+use web5::apid::dids::did::DidError;
+use web5::apid::dids::methods::MethodError;
+use web5::apid::dsa::DsaError;
+use web5::apid::{in_memory_key_manager::KeyManagerError, jwk::JwkError};
+
+#[derive(Debug, Error)]
+pub enum RcbError {
+    #[error("{message}")]
+    Error {
+        message: String,
+        error_type: String,
+        error_variant: String,
+    },
+}
+
+impl RcbError {
```

Comment:
I dig it, this is a wrapper to all the other errors right?

With the kotlin code be able to determine it's type right by doing type_of?

Reply:
yup! went with `RustCoreError` instead

Reply:
also notice I slightly changed `error_variant()` to just `variant()` 

I wanted to do the same with `error_type()` but I'm concerned `type()` being a commonly used special word in programming languages that would cause issues